### ISSUE TITLE
co-location not working in addon?

### ISSUE THREAD

**Issue by MichalBryxi** [ember-cli/ember-cli-htmlbars#337]
Following testing repository: https://gitlab.com/michal.bryxi/ember-octane-colocating-in-addon
Uses: following testing branch of my addon: https://gitlab.com/michal.bryxi/ember-safe-button/tree/ember-octane-bug-report

This addon uses:
```
"dependencies": {
    "ember-cli-htmlbars": "^4.0.7",
}
```

With following structure (co-located components):
```
❯ tree addon
addon
├── components
│   ├── safe-button.hbs
│   ├── safe-button.js
│   ├── safety.hbs
│   ├── safety.js
│   ├── trigger.hbs
│   └── trigger.js
├── index.js
```

When trying to install given addon to my testing app and run it, I will get:
```
❯ ember s
DEPRECATION: ember-cli-htmlbars-inline-precompile is no longer needed with ember-cli-htmlbars versions 4.0.0 and higher, please remove it from `/Users/mbryxi/ember/ember-octane-colocating-in-addon/package.json`
Build Error (broccoli-persistent-filter:TemplateCompiler)

EEXIST: file already exists, symlink '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-190-funnel_funnel_ember_safe_button_addon/ember-safe-button/components/safe-button.js' -> '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-191-broccoli_persistent_filter_template_compiler/ember-safe-button/components/safe-button.js'


Stack Trace and Error Report: /var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/error.dump.079f3756e030a7788eb52ed49fcb0b79.log
```

Mentioned log contains:
```
❯ cat /var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/error.dump.079f3756e030a7788eb52ed49fcb0b79.log
=================================================================================

ENV Summary:

  TIME: Fri Oct 18 2019 22:36:02 GMT+0100 (British Summer Time)
  TITLE: ember
  ARGV:
  - /Users/mbryxi/.volta/tools/image/node/10.16.3/6.9.0/bin/node
  - /Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/.bin/ember
  - s
  EXEC_PATH: /Users/mbryxi/.volta/tools/image/node/10.16.3/6.9.0/bin/node
  TMPDIR: /var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T
  SHELL: /bin/zsh
  PATH:
  - /Users/mbryxi/.volta/tools/image/node/10.16.3/6.9.0/bin
  - /Users/mbryxi/.volta/tools/image/yarn/1.19.0/bin
  - /usr/local/opt/ruby/bin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  - /usr/local/opt/fzf/bin
  PLATFORM: darwin x64
  FREEMEM: 143097856
  TOTALMEM: 17179869184
  UPTIME: 220463
  LOADAVG: 3.1240234375,5.24853515625,5.025390625
  CPUS:
  - Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz - 2700
  - Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz - 2700
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.15.0
  - brotli: 1.0.7
  - cldr: 35.1
  - http_parser: 2.8.0
  - icu: 64.2
  - modules: 64
  - napi: 4
  - nghttp2: 1.39.2
  - node: 10.16.3
  - openssl: 1.1.1c
  - tz: 2019a
  - unicode: 12.1
  - uv: 1.28.0
  - v8: 6.8.275.32-node.54
  - zlib: 1.2.11

ERROR Summary:

  - broccoliBuilderErrorStack: Error: EEXIST: file already exists, symlink '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-190-funnel_funnel_ember_safe_button_addon/ember-safe-button/components/safe-button.js' -> '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-191-broccoli_persistent_filter_template_compiler/ember-safe-button/components/safe-button.js'
    at Object.symlinkSync (fs.js:909:3)
    at symlink (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/symlink-or-copy/index.js:118:14)
    at symlinkOrCopySync (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/symlink-or-copy/index.js:85:5)
    at resolve (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:322:16)
    at initializePromise (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/rsvp/dist/rsvp.js:523:7)
    at new Promise (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/rsvp/dist/rsvp.js:1017:35)
    at TemplateCompiler.Filter._handleFile (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:306:10)
    at createOperation (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:277:25)
    at patch (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:283:18)
    at /Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/promise-map-series/index.js:11:14
  - code: [undefined]
  - codeFrame: EEXIST: file already exists, symlink '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-190-funnel_funnel_ember_safe_button_addon/ember-safe-button/components/safe-button.js' -> '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-191-broccoli_persistent_filter_template_compiler/ember-safe-button/components/safe-button.js'
  - errorMessage: EEXIST: file already exists, symlink '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-190-funnel_funnel_ember_safe_button_addon/ember-safe-button/components/safe-button.js' -> '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-191-broccoli_persistent_filter_template_compiler/ember-safe-button/components/safe-button.js'
        at broccoli-persistent-filter:TemplateCompiler
-~- created here: -~-
    at TemplateCompiler.Plugin (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-plugin/index.js:7:31)
    at new Filter (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:87:10)
    at new TemplateCompiler (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-htmlbars/lib/template-compiler-plugin.js:35:5)
    at Object.toTree (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-htmlbars/lib/ember-addon-main.js:81:26)
    at /Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-preprocess-registry/preprocessors.js:188:26
    at Array.forEach (<anonymous>)
    at processPlugins (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-preprocess-registry/preprocessors.js:186:11)
    at module.exports.preprocessTemplates (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-preprocess-registry/preprocessors.js:169:10)
    at Class.compileTemplates (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli/lib/models/addon.js:1222:33)
    at Class.compileAddon (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli/lib/models/addon.js:1288:42)
-~- (end) -~-
  - errorType: Build Error
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
    - treeDir: [undefined]
  - message: EEXIST: file already exists, symlink '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-190-funnel_funnel_ember_safe_button_addon/ember-safe-button/components/safe-button.js' -> '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-191-broccoli_persistent_filter_template_compiler/ember-safe-button/components/safe-button.js'
        at broccoli-persistent-filter:TemplateCompiler
-~- created here: -~-
    at TemplateCompiler.Plugin (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-plugin/index.js:7:31)
    at new Filter (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:87:10)
    at new TemplateCompiler (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-htmlbars/lib/template-compiler-plugin.js:35:5)
    at Object.toTree (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-htmlbars/lib/ember-addon-main.js:81:26)
    at /Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-preprocess-registry/preprocessors.js:188:26
    at Array.forEach (<anonymous>)
    at processPlugins (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-preprocess-registry/preprocessors.js:186:11)
    at module.exports.preprocessTemplates (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli-preprocess-registry/preprocessors.js:169:10)
    at Class.compileTemplates (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli/lib/models/addon.js:1222:33)
    at Class.compileAddon (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/ember-cli/lib/models/addon.js:1288:42)
-~- (end) -~-
  - name: BuildError
  - nodeAnnotation: [undefined]
  - nodeName: broccoli-persistent-filter:TemplateCompiler
  - originalErrorMessage: EEXIST: file already exists, symlink '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-190-funnel_funnel_ember_safe_button_addon/ember-safe-button/components/safe-button.js' -> '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-191-broccoli_persistent_filter_template_compiler/ember-safe-button/components/safe-button.js'
  - stack: Error: EEXIST: file already exists, symlink '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-190-funnel_funnel_ember_safe_button_addon/ember-safe-button/components/safe-button.js' -> '/var/folders/dw/bx1n948x7bgdmwdpxgns8q040000gn/T/broccoli-930Nhgx0466jXRs/out-191-broccoli_persistent_filter_template_compiler/ember-safe-button/components/safe-button.js'
    at Object.symlinkSync (fs.js:909:3)
    at symlink (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/symlink-or-copy/index.js:118:14)
    at symlinkOrCopySync (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/symlink-or-copy/index.js:85:5)
    at resolve (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:322:16)
    at initializePromise (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/rsvp/dist/rsvp.js:523:7)
    at new Promise (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/rsvp/dist/rsvp.js:1017:35)
    at TemplateCompiler.Filter._handleFile (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:306:10)
    at createOperation (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:277:25)
    at patch (/Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/broccoli-persistent-filter/index.js:283:18)
    at /Users/mbryxi/ember/ember-octane-colocating-in-addon/node_modules/promise-map-series/index.js:11:14

=================================================================================
```

**Comment by rwjblue** [comment]
Thank you for reporting! I'll try to dig into this repro and see what needs to be done to fix...

**Comment by rwjblue** [comment]
OK, so I poked at this a bit. The error is being caused by the fact that the addon is relying on colocation (during compilation), but ember-cli-htmlbars is still currently **only** doing colocation based compilation when the host application has opted in to use octane (e.g. `setEdition('octane')` in its `.ember-cli.js` file).

**Comment by rwjblue** [comment]
Just released ember-cli-htmlbars@4.2.0 which supports colocation regardless of the host applications octane status. The only requirements are:

* The addon is using `ember-cli-babel` >= 7.11.0
* The application is using `ember-cli` >= 3.12.0-beta.2
* The application is using `ember-source` >= 3.13.0

Closing...
